python/dataproc_templates/pubsublite/pubsublite_to_bigquery.py

```diff
@@ -0,0 +1,161 @@
+            required=True,
+            help='BigQuery output dataset'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_OUTPUT_TABLE}',
+            dest=constants.PUBSUBLITE_TO_BQ_OUTPUT_TABLE,
+            required=True,
+            help='BigQuery output table'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_TO_BQ_TEMPORARY_BUCKET}',
+            dest=constants.PUBSUBLITE_TO_BQ_TEMPORARY_BUCKET,
+            required=True,
+            help='Temporary bucket for the Spark BigQuery connector'
+        )
+        parser.add_argument(
+            f'--{constants.PUBSUBLITE_CHECKPOINT_LOCATION}',
+            dest=constants.PUBSUBLITE_CHECKPOINT_LOCATION,
+            required=True,
+            help='Temporary folder for checkpoint location'
+        )
+
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+        logger: Logger = self.get_logger(spark=spark)
+
+        # Arguments
+        input_subscription: str = args[constants.PUBSUBLITE_TO_BQ_INPUT_SUBSCRIPTION]
+        output_project_id: str = args[constants.PUBSUBLITE_TO_BQ_PROJECT_ID]
+        output_dataset: str = args[constants.PUBSUBLITE_TO_BQ_OUTPUT_DATASET]
+        output_table: str = args[constants.PUBSUBLITE_TO_BQ_OUTPUT_TABLE]
+        pubsublite_checkpoint_location: str = args[constants.PUBSUBLITE_CHECKPOINT_LOCATION]
+        bq_temp_bucket: str = args[constants.PUBSUBLITE_TO_BQ_TEMPORARY_BUCKET]
+        timeout_ms: int = args[constants.PUBSUBLITE_TO_BQ_INPUT_TIMEOUT_MS]
+
+        logger.info(
+            "Starting Pubsublite to Bigquery spark job with parameters:\n"
+            f"{pprint.pformat(args)}"
+        )
+        # Set configuration to connect to Pubsublite by overwriting the spark session
+        spark = (
+            SparkSession
+                .builder
+                .appName("read-app")
+                .master("yarn")
+                .getOrCreate())
+
+        # Read
+        input_data=(spark.readStream \
+            .format(constants.FORMAT_PUBSUBLITE) \
+            .option(f"{constants.FORMAT_PUBSUBLITE}.subscription", input_subscription,) \
+            .load())
+        
+        input_data.withColumn("data", input_data.data.cast(StringType()))
+
+        # Write
+        query = (input_data.writeStream \
+            .format(constants.FORMAT_BIGQUERY) \
+            .option("temporaryGcsBucket", bq_temp_bucket) \
+            .option("checkpointLocation", pubsublite_checkpoint_location) \
+            .option("table", f"{output_project_id}.{output_dataset}.{output_table}") \
+            .trigger(processingTime="1 second") \
+            .start())
+
+        # Wait timeout_ms/1000 seconds (must be >= 60 seconds) to start receiving messages.
+        query.awaitTermination(timeout_ms/1000)
```

Comment:
Why are we doing this? Can't we specify to provide timeout in seconds?

Reply:
Please update example submission. I do not think this property is being specified there.

Reply:
Done